include/grpcpp/security/credentials.h

```diff
@@ -122,22 +109,29 @@ class ChannelCredentials : private grpc::internal::GrpcLibrary {
 /// \see https://grpc.io/docs/guides/auth.html
 class CallCredentials : private grpc::internal::GrpcLibrary {
  public:
+  ~CallCredentials() override;
+
   /// Apply this instance's credentials to \a call.
-  virtual bool ApplyToCall(grpc_call* call) = 0;
-  virtual grpc::string DebugString() {
-    return "CallCredentials did not provide a debug string";
-  }
+  virtual bool ApplyToCall(grpc_call* call);
+
+  virtual grpc::string DebugString();
 
  protected:
   friend std::shared_ptr<ChannelCredentials> CompositeChannelCredentials(
       const std::shared_ptr<ChannelCredentials>& channel_creds,
       const std::shared_ptr<CallCredentials>& call_creds);
-
   friend std::shared_ptr<CallCredentials> CompositeCallCredentials(
       const std::shared_ptr<CallCredentials>& creds1,
       const std::shared_ptr<CallCredentials>& creds2);
+  friend std::string grpc::testing::GetOauth2AccessToken();
+  friend std::shared_ptr<CallCredentials> MakeCallCredentials(
```

Comment:
Could avoid this by using a subclass pattern, similar to what I suggested elsewhere for `InsecureServerCredentials`.

Could also remove the friend statement for `CompositeCallCredentials()` with the same pattern.